crates/primitives/src/block.rs

```diff
@@ -83,6 +100,38 @@ impl Deref for SealedBlock {
     }
 }
 
+/// Sealed block with senders recovered from transactions.
+#[derive(Debug, Clone, PartialEq, Eq, Default)]
+pub struct SealedBlockWithSenders {
+    /// Sealed block
+    pub block: SealedBlock,
+    /// List of senders that match trasanctions from block.
+    pub senders: Vec<Address>,
+}
+
+impl SealedBlockWithSenders {
+    /// New sealed block with sender. Return none if len of tx and senders does not match
+    pub fn new(block: SealedBlock, senders: Vec<Address>) -> Option<Self> {
```

Comment:
Should return an error instead

Reply:
Not sure what exact error type should i return, if there is only one "error" case. And this is similar what other functions for tx and block are doing.